ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -110,20 +108,22 @@ export class ScanComponent implements OnInit {
 
   private validateValue(labelValue: string, position: number, isLeft: boolean): boolean {
     let isDuplicate = false;
-    for (let i = 0; i < this.scanPairsValue.length; i++) {
-      if (i !== position) {
-        if (this.scanPairsValue[position].leftValue != null && labelValue === this.scanPairsValue[i].leftValue
-          || this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
-          isDuplicate = true;
-        }
-      } else {
-        if (isLeft) {
-          if (this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
+    if(labelValue !== '') {
+      for (let i = 0; i < this.scanPairsValue.length; i++) {
+        if (i !== position) {
+          if (this.scanPairsValue[position].leftValue != null && labelValue === this.scanPairsValue[i].leftValue
+            || this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
```

Comment:
This conditional check is too verbose and hard to read.
I would suggest extracting this logic into a separate function and calling that function in `if()`
Like so:
```
private validateValue(...) {
...
  if( iAmNotNullAndBlaBla() ) {
  ...
  }
...
}

private iAmNotNullAndBlaBla() {
...
}
```

Reply:
I added a helper function as you asked to make the if statement a bit easier to follow.